### ISSUE TITLE
add icon for uMatrix

### ISSUE THREAD

**Issue by collinbarrett** [collinbarrett/FilterLists#688]
@collinbarrett please add an icon to it for the field - Software

PS Although the umatrix icon is scary. forgive me gorhill

_Originally posted by @bogachenko in https://github.com/collinbarrett/FilterLists/issues/164#issuecomment-462870888_

**Comment by DandelionSprout** [comment]
I have a good image for uMatrix (below) that could be used, as well as for five other software tools as well. But when the pull I made for it broke in a pretty bizarre way, I haven't felt like making another shot at it myself...

![image](https://user-images.githubusercontent.com/22780683/52730507-d9bddf80-2fbb-11e9-9569-dbb25575a2c9.png)

**Comment by DandelionSprout** [comment]
For the record, here's ones for Hosts File Editor, Google Hit Hider by Domain, Diversion, dnsmasq, and Slimjet as well.

![11-hosts-file-editor](https://user-images.githubusercontent.com/22780683/52730696-4a64fc00-2fbc-11e9-8881-bf30a39f633f.png) 
![17-google-hit-hider-by-domain](https://user-images.githubusercontent.com/22780683/52730727-594bae80-2fbc-11e9-9414-08e6666c7c21.png)
![22-diversion](https://user-images.githubusercontent.com/22780683/52730728-5a7cdb80-2fbc-11e9-9b70-c9bfce658cf8.png)
![23-dnsmasq](https://user-images.githubusercontent.com/22780683/52730731-5bae0880-2fbc-11e9-9625-eeb2ea33b8ad.png)
![24-slimjet](https://user-images.githubusercontent.com/22780683/52730739-5d77cc00-2fbc-11e9-9310-c61eaf4c5c16.png)

**Comment by DandelionSprout** [comment]
The images are uncompressed, but that was the least of my worries at the time.

**Comment by bogachenko** [comment]
> 
> 
> I have a good image for uMatrix (below) that could be used, as well as for five other software tools as well. But when the pull I made for it broke in a pretty bizarre way, I haven't felt like making another shot at adding them myself...
> 
> ![image](https://user-images.githubusercontent.com/22780683/52730507-d9bddf80-2fbb-11e9-9569-dbb25575a2c9.png)

my northern friend ... why do you repeat everything? in ur pull https://github.com/collinbarrett/FilterLists/pull/661/files

![default](https://user-images.githubusercontent.com/8535285/52750912-ab82e300-3041-11e9-9e54-773323111a67.png)

mistake in file SoftwareIcon.tsx (https://github.com/DandelionSprout/FilterLists/blob/2e2c68d8c05ec923c0eaa64c23109cdb7a474e3c/src/FilterLists.Web/ClientApp/modules/home/components/softwareIcon/SoftwareIcon.tsx)
 
you have so
```
import * as React from "react";
import {
    img1,
    img2,
    img3,
    img4,
    img5,
    img6,
    img7,
    img8,
    img10,
    img11,
    img12,
    img13,
    img14,
    img15,
    img16,
    img17,
    img18,
    img19,
    img20,
    img21,
    img22,
    img23,
    img24,
    img25
} from "./imgs";

interface IProps {
    id: number;
};

export const SoftwareIcon = (props: IProps) =>
    icons[props.id]
    ? <img src={icons[props.id].image}
           width="20"
           alt={icons[props.id].imageTitle}
           title={icons[props.id].imageTitle}/>
    : null;

interface IIcon {
    image: any;
    imageTitle: string;
};

const icons: { [id: number]: IIcon; } = {
    1: { image: img1, imageTitle: "uBlock Origin" },
    2: { image: img2, imageTitle: "Adblock Plus" },
    3: { image: img3, imageTitle: "AdGuard" },
    4: { image: img4, imageTitle: "DNS66" },
    5: { image: img5, imageTitle: "Nano Adblocker" },
    6: { image: img6, imageTitle: "AdBlock" },
    7: { image: img7, imageTitle: "AdAway" },
    8: { image: img8, imageTitle: "Personal Blocklist" },
    10: { image: img10, imageTitle: "Redirector" },
    11: { image: img10, imageTitle: "Hosts File Editor" },
    12: { image: img12, imageTitle: "Gas Mask" },
    13: { image: img13, imageTitle: "MinerBlock" },
    14: { image: img14, imageTitle: "Pi-hole" },
    15: { image: img15, imageTitle: "uBlock" },
    16: { image: img16, imageTitle: "Internet Explorer (TPL)" },
    17: { image: img10, imageTitle: "Google Hit Hider by Domain" },
    18: { image: img18, imageTitle: "FireHOL" },
    19: { image: img19, imageTitle: "Samsung Knox" },
    20: { image: img20, imageTitle: "Little Snitch" },
    21: { image: img21, imageTitle: "Privoxy" },
    22: { image: img20, imageTitle: "Diversion" },
    23: { image: img20, imageTitle: "dnsmasq" },
    24: { image: img20, imageTitle: "Slimjet" },
    25: { image: img20, imageTitle: "uMatrix" }
};
```

it should be like that!

```
import * as React from "react";
import {
    img1,
    img2,
    img3,
    img4,
    img5,
    img6,
    img7,
    img8,
    img10,
    img11,
    img12,
    img13,
    img14,
    img15,
    img16,
    img17,
    img18,
    img19,
    img20,
    img21,
    img22,
    img23,
    img24,
    img25
} from "./imgs";

interface IProps {
    id: number;
};

export const SoftwareIcon = (props: IProps) =>
    icons[props.id]
    ? <img src={icons[props.id].image}
           width="20"
           alt={icons[props.id].imageTitle}
           title={icons[props.id].imageTitle}/>
    : null;

interface IIcon {
    image: any;
    imageTitle: string;
};

const icons: { [id: number]: IIcon; } = {
    1: { image: img1, imageTitle: "uBlock Origin" },
    2: { image: img2, imageTitle: "Adblock Plus" },
    3: { image: img3, imageTitle: "AdGuard" },
    4: { image: img4, imageTitle: "DNS66" },
    5: { image: img5, imageTitle: "Nano Adblocker" },
    6: { image: img6, imageTitle: "AdBlock" },
    7: { image: img7, imageTitle: "AdAway" },
    8: { image: img8, imageTitle: "Personal Blocklist" },
    10: { image: img10, imageTitle: "Redirector" },
    11: { image: img11, imageTitle: "Hosts File Editor" },
    12: { image: img12, imageTitle: "Gas Mask" },
    13: { image: img13, imageTitle: "MinerBlock" },
    14: { image: img14, imageTitle: "Pi-hole" },
    15: { image: img15, imageTitle: "uBlock" },
    16: { image: img16, imageTitle: "Internet Explorer (TPL)" },
    17: { image: img17, imageTitle: "Google Hit Hider by Domain" },
    18: { image: img18, imageTitle: "FireHOL" },
    19: { image: img19, imageTitle: "Samsung Knox" },
    20: { image: img20, imageTitle: "Little Snitch" },
    21: { image: img21, imageTitle: "Privoxy" },
    22: { image: img22, imageTitle: "Diversion" },
    23: { image: img23, imageTitle: "dnsmasq" },
    24: { image: img24, imageTitle: "Slimjet" },
    25: { image: img25, imageTitle: "uMatrix" }
};

```

 Maybe this is a mistake?

**Comment by DandelionSprout** [comment]
Wait, how come no one told me this before and/or figured it out? This is actually super-great information, and I'm surprised I never spotted that.

You (Bogachenko) normally strike me as a somewhat silly guy, but that was some splendid detective work you've done, and you have my standing applauses for it. 👏 

I'll now definitely do a 2nd attempt at the pull once I get home from work sometime tomorrow.

**Comment by collinbarrett** [comment]
Agreed. Good catch, @bogachenko . Thanks.

Sorry, ya'll. My time has been so limited. I really appreciate the contributions ya'll are making, and I've been trying to keep up with those, but my quality control and further development has been very limited as of late.

I hope for that to change...

**Comment by bogachenko** [comment]
@DandelionSprout 

>silly guy

okey.

**Comment by DandelionSprout** [comment]
Wait, that was not what I meant.

I was referring to things like calling me John Snow, the sometimes interestingly named pull titles, and thinking of Travis as an addressable entity which are uncommon in tech projects. They're not inherently negative, but has made me slightly uncertain how to discuss things with you.

I can be a playful fun-seeker too, but I tend to be a bit more straight-forward when it comes to work-like things I'm involved in. Hopefully that's understandable and/or forgivable.

**Comment by collinbarrett** [comment]
closed by #692

**Comment by bogachenko** [comment]
@DandelionSprout what you wrote is so ridiculous, I don’t know where to start..

> calling me John Snow

it's just a nickname, a banal nickname. I wanted to draw a parallel between what nationality you are and where you come from with the hero of the series, who has a talking last name. In Russia, people give nicknames, and I'm sure that in Western culture as well, even if you take into account the site on which I am now. (for example there is a githab repository "ghacks-user.js
"it was created by a dude named Thorin-Oakenpants. The end of his last name - "pants", When people turn to him they don't write his name, they put smiley 👖 , or write pants, that's his the nickname, no one takes offense and it's funny (for me))

Ha, but I find it (АХУЕННО БЛЯТЬ) funny the fact that you couldn't write me something like:

- Ivan (or Boris or Nikolay) you made fuck ? I don't like that nickname, I don't like nicknames at all. Stop it!

and I am:

- Okay, no problem.

> the sometimes interestingly named commit titles
I don't like documentation, it annoys me. So I deliberately do not fill it. It is impossible to leave these fields blank, there need to write something...

> thinking of the uMatrix icon as "scary"

here I have no comments at all, I do not know how to comment on it...
I don't like the uMatrix icon, is it scary (for me)? When I wrote the post I wanted to offer to give up the icons at all, instead, make text, but then I thought that it was a stupid idea, I refused it and wiped. But left my opinion about this icon, I don't know why. Just? Just like that?

How can I comment on this? my god.

> thinking of Travis as an addressable entity

returning to the commit text. Since it was called a bot by a HUMAN name, in the commit field I write something like " Baby Travis let's go".
I need to write something there. Is that stupid? Okay. ha. as you say.

> which are uncommon in tech projects.

apparently you and I are sitting on different projects that do not fit your criteria

> They're not inherently negative, but has made me slightly uncertain how to discuss things with you.

Of course, you don't need to talk about the problem. Just ... just be patient )


and text in the pull

> If @bogachenko is willing to forgive me for something that wasn't even an actual insult in the first place (I'm pretty bad at diplomacy), then he has my thanks once more for pointing me to this fix

> If @bogachenko is willing to forgive me for something that wasn't even an actual insult in the first place

If this isn't an insult to you, what are you sorry for?
Who's the scary person that makes you do this? Show me? If you call a person stupid ass (ya ya ya cut to the root, what you want to say, you didn't say that, but it's tantamount to) is like a handshake, why do you apologize against your will?

So. Like I said, relationships are ruined. I will be patient and silent towards you. Dot. In Russia, there is a proverb that is very suitable for this situation:

Дурак умен, пока молчит => The smart fool is silent.

To you I'll be silent, do not want to infect you with stupidity or naivete or fuck knows what.

lucky day
